lighthouse-core/audits/third-party-summary.js

```diff
@@ -110,39 +200,43 @@ class ThirdPartySummary extends Audit {
     const multiplier = settings.throttlingMethod === 'simulate' ?
       settings.throttling.cpuSlowdownMultiplier : 1;
 
-    const summaryByEntity = ThirdPartySummary.getSummaryByEntity(networkRecords, tasks, multiplier);
-
-    const summary = {wastedBytes: 0, wastedMs: 0};
+    const summaries = ThirdPartySummary.getSummaries(networkRecords, tasks, multiplier);
+    const overallSummary = {wastedBytes: 0, wastedMs: 0};
 
-    const results = Array.from(summaryByEntity.entries())
+    const results = Array.from(summaries.byEntity.entries())
       // Don't consider the page we're on to be third-party.
       // e.g. Facebook SDK isn't a third-party script on facebook.com
       .filter(([entity]) => !(mainEntity && mainEntity.name === entity.name))
       .map(([entity, stats]) => {
-        summary.wastedBytes += stats.transferSize;
-        summary.wastedMs += stats.blockingTime;
+        overallSummary.wastedBytes += stats.transferSize;
+        overallSummary.wastedMs += stats.blockingTime;
 
         return {
+          ...stats,
           entity: /** @type {LH.Audit.Details.LinkValue} */ ({
             type: 'link',
             text: entity.name,
             url: entity.homepage || '',
           }),
-          transferSize: stats.transferSize,
-          mainThreadTime: stats.mainThreadTime,
-          blockingTime: stats.blockingTime,
+          subItems: /** @type {LH.Audit.Details.TableSubItems} */ ({
+            type: 'subitems',
+            items: ThirdPartySummary.getSubItems(entity, summaries, stats),
+          }),
         };
       })
       // Sort by blocking time first, then transfer size to break ties.
       .sort((a, b) => (b.blockingTime - a.blockingTime) || (b.transferSize - a.transferSize));
 
     /** @type {LH.Audit.Details.Table['headings']} */
     const headings = [
-      {key: 'entity', itemType: 'link', text: str_(UIStrings.columnThirdParty)},
+      {key: 'entity', itemType: 'link', text: str_(UIStrings.columnThirdParty),
```

Comment:
prefer one line per heading but disable eslint (see lighthouse-core/audits/byte-efficiency/duplicated-javascript.js for example)

Reply:
(I see this is an existing format, but a change would be appreciated)